### PR TITLE
Add new case of attaching iface with coalesce

### PR DESCRIPTION
**Pull request by chloerh**
- RHEL7-91214 - [attach-device][coalesce] hotplug with coalesce
 settings in interface, then save and restore

Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
Depends on https://github.com/avocado-framework/avocado-vt/pull/3005

**Comment by chloerh** [comment]
JOB ID     : ed2377aa0c477972c0d8740c54cbcb359d5fa3af
JOB LOG    : /root/avocado/job-results/job-2021-04-15T19.29-ed2377a/job.log
 (1/1) type_specific.local.virsh.attach_detach_interface.normal_test.vm_running.default_network.with_coalesce.domname: PASS (41.44 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 41.74 s

**Comment by chloerh** [comment]
@yalzhang Would you please help review?

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py [hunk 1]
```diff
@@ -192,6 +192,32 @@ def add_pcie_controller(vm_name):
...
+    iface_features = vm_xml.VMXML.get_iface_by_mac(vm_name, mac)
+    target = iface_features.get('target', {}).get('dev')
+
+    frames = eval(coalesce).get('max')
+
+    ethtool_cmd = 'ethtool -c %s' % target
+    ethtool_output = process.run(
+        ethtool_cmd, verbose=True, ignore_status=True).stdout_text
+
+    expected_str = 'rx-frames: %s'
```
Comment (kylazhang) [inline libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py:213]: I think here should be 
```
expected_str = 'rx-frames: %s' % frames
```
and also need to update the following if condition param?

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py [hunk 2]
```diff
@@ -443,6 +476,12 @@ def run(test, params, env):
         if save_restore == "yes":
             check_save_restore(vm_name)
 
+        # Check coalesce info if needed after save/restore
+        if iface_coalesce:
+            stat, ret = check_coalesce(vm_name, iface_mac, iface_coalesce)
+            fail_flag = fail_flag | stat
+            result_info.append(ret)
```
Comment (kylazhang) [inline libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py:483]: Seems same with above check part? Could we merge them to 1 function?

**Review by kylazhang — COMMENTED** [review]
Please check the comments inline, thanks

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py (hunk 1 above)
Comment (chloerh) [reply libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py:213]: Agreed. :joy: Actually I don't know why I wrote it that way. Updated.

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py (hunk 2 above)
Comment (chloerh) [reply libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py]: You're right! I just updated.

**Comment by chloerh** [comment]
New result:
PASS 1-type_specific.local.virsh.attach_detach_interface.normal_test.vm_running.default_network.with_coalesce.domname

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py (hunk 2 above)
Comment (chloerh) [reply libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py]: Updated. At first I used nonlocal which cannot pass travis check of python2.7. Now it should work.

**Review by kylazhang — APPROVED** [review]
LGTM

**Review by yalzhang — APPROVED** [review]

**Pull request merged**